metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -973,11 +1176,40 @@ def get_schema_metadata_for_custom_sql(
         )
         return schema_metadata
 
-    def _get_project(self, node):
-        if node.get("__typename") == "EmbeddedDatasource" and node.get("workbook"):
-            return node["workbook"].get("projectName")
-        elif node.get("__typename") == "PublishedDatasource":
-            return node.get("projectName")
+    def _get_project(self, ds):
+        if self.config.extract_project_hierarchy:
+            if (
+                ds.get("__typename") == "PublishedDatasource"
+                and ds.get("luid")
+                and ds["luid"] in self.datasource_project_map.keys()
+                and self.datasource_project_map[ds["luid"]]
```

Comment:
this accessor logic is a bit hard to follow/ maybe comments would help?

also, can we use constants for luid? and PublishedDatasource?

Reply:
@jjoyce0510 Can we skip the comment related to constant and take it as part of refactoring PR.

Added comments in code

Reply:
yes